cdc/log.cc

```diff
@@ -234,7 +237,29 @@ static schema_ptr create_log_schema(const schema& s, std::optional<utils::UUID>
         for (const auto& column : columns) {
             auto type = column.type;
             if (is_data_col) {
-                type = tuple_type_impl::get_instance({ /* op */ data_type_for<column_op_native_type>(), /* value */ type, /* ttl */long_type});
+                type = visit(*type, make_visitor(
+                    // lists are represented as map<timeuuid, value_type>. Otherwise we cannot express delta
+                    [] (const list_type_impl& type) {
+                        return map_type_impl::get_instance(type.name_comparator(), type.value_comparator(), false);
+                    },
+                    // user types are expressed as tuple<tuple<field0>, tuple<field1>...>
+                    // the extra tuple allows us to distinguish null == <no info>
+                    // from <null> = set field to null
+                    [] (const user_type_impl& type) -> data_type {
+                        std::vector<data_type> types;
+                        types.reserve(type.size());
+                        for (auto& ft : type.field_types()) {
+                            types.emplace_back(tuple_type_impl::get_instance({ ft }));
+                        }
+                        return tuple_type_impl::get_instance(std::move(types));
```

Comment:
Will this be interpretable after the original UDT has been altered? Or is that not even a goal?

Reply:
No, and no. 
Longer answer: It is interpretable more or less the same way as it is now: You need to know what field 0, 1, 2 etc are in the consumer. The schema here will tell you field type, but no semantic info. Could maybe make the structure tuple<name, value>, but that seems expensive. 
I think it is "buyer beware" when using any non-atomics with cdc. Esp. udt.